linked_list/main.py

```diff
@@ -62,34 +68,116 @@ class LinkedList(AbstractLinkedList):
     """
 
     def __init__(self, elements=None):
-        pass
+        self.start = None
+        self.end = None
+        
+        if elements:
+            
+            self.end = Node(elements[-1])
+            node = Node(elements[0], None)
+            last_node = self.end
+            for element in reversed(elements[:-1]):
```

Comment:
Good idea going backwards through the list. Our solution entailed keeping track of indexes and some flow control, which always gets messy.